core/components/atoms/text/Text.tsx

```diff
@@ -41,7 +45,8 @@ export const Text = (props: TextProps) => {
     Text: true,
     [`Text--${appearance}`]: appearance,
     [`Text--${weight}`]: weight,
-    ['Text--small']: small,
+    [`Text--${size}`]: size,
+    ['Text--small']: size === 'small' || small,
```

Comment:
If this is just for backward compatibility for now, put a comment over here to remove this later.